components/kyma-environment-broker/internal/hyperscaler/azure/client.go

```diff
@@ -42,11 +65,57 @@ func (nc *AzureClient) CreateNamespace(ctx context.Context, azureCfg *Config, gr
 	// we need to use a copy of the location, because the following azure call will modify it
 	locationCopy := azureCfg.GetLocation()
 	parameters := eventhub.EHNamespace{Location: &locationCopy, Tags: tags}
-	ehNamespace, err := nc.createAndWait(ctx, groupName, namespace, parameters)
+	ehNamespace, err := nc.createAndWaitNamespace(ctx, groupName, namespace, parameters)
 	return &ehNamespace, err
 }
 
-func (nc *AzureClient) createAndWait(ctx context.Context, resourceGroupName string, namespaceName string, parameters eventhub.EHNamespace) (result eventhub.EHNamespace, err error) {
+func (nc *AzureClient) DeleteResourceGroup(ctx context.Context, tags Tags) (resources.GroupsDeleteFuture, error) {
```

Comment:
In the underlying `Future` documentation they said:
```
// Future provides a mechanism to access the status and results of an asynchronous request.
// Since futures are stateful they should be passed by value to avoid race conditions.
type Future struct {
	pt pollingTracker
}
```
I think passing the wrapping struct by value is safer. What do you think?